gson/src/main/java/com/google/gson/Gson.java

```diff
@@ -487,10 +489,12 @@ static void checkValidFloatingPoint(double value) {
    */
   public <T> TypeAdapter<T> getDelegateAdapter(TypeAdapterFactory skipPast, TypeToken<T> type) {
     boolean skipPastFound = false;
```

Comment:
this is only used in the loop below, can we move the `if` above this declaration